src/PooledArrays.jl

```diff
@@ -315,8 +315,8 @@ Base.getindex(A::PooledArray, I::AbstractArray) =
 
 function getpoolidx{T,R}(pa::PooledArray{T,R}, val::Any)
     val::T = convert(T,val)
-    pool_idx = findfirst(pa.pool, val)
-    if pool_idx <= 0
+    pool_idx = searchsortedfirst(pa.pool, val)
+    if pool_idx > length(pa.pool) || pa.pool[i] != val
```

Comment:
aw crap. I ran the test locally on the wrong Julia version, beats me why I used `i`.